tests/acceptance/integration-test.js

```diff
@@ -52,3 +52,42 @@ test('sticky messages are not removed automatically', function(assert) {
     assert.equal(find('.alert.alert-danger p').text(), 'You went offline');
   });
 });
+test('sticky messages are still removed when clicked', function(assert) {
+  assert.expect(5);
+  
+  visit('/');
+
+  andThen(() => {
+    assert.ok(find('.alert.alert-danger'));
+    assert.equal(find('.alert').length, 1);
+    assert.equal(find('.alert.alert-danger h6').text(), 'Danger');
+    assert.equal(find('.alert.alert-danger p').text(), 'You went offline');
+  });
+  
+  // click('.alert.alert-danger');
+  triggerEvent('.alert.alert-danger', $(document), 'click');
+  andThen(() => {
+    assert.equal(find('.alert').length, 0);
+  });
+});
+
+test('mouseover-ed messages are not removed automatically', function (assert) {
+  assert.expect(8);
+  visit('/');
+  triggerEvent('.alert.alert-success', 'mouseenter');
+  
+  assert.ok(find('.alert.alert-success'), 'right after the rendering the flash message is there!');
+  andThen(() => {
+    assert.ok(find('.alert.alert-success'), 'the success message received the mouseenter and stayed on display');
+    assert.equal(find('.alert').length, 2); //there should be the sticky and the mouseover-ed
+    assert.equal(find('.alert.alert-success h6').text(), 'Success');
+    assert.equal(find('.alert.alert-success p').text(), 'Route transitioned successfully');
+  });
+  triggerEvent('.alert.alert-success', 'mouseleave');
```

Comment:
Try moving this below `L84`. The issue is probably that the `triggerEvent` call happens synchronously before the `andThen` calls can finish